var/spack/repos/builtin/packages/ispc/package.py

```diff
@@ -67,11 +67,12 @@ class Ispc(CMakePackage):
         sha256="d3ccf547d3ba59779fd375e10417a436318f2200d160febb9f830a26f0daefdc",
     )
 
-    def setup_build_environment(self, env):
-        if self.spec.satisfies("@1.18.0:"):
-            env.append_flags("LDFLAGS", "-lcurses")
-            env.append_flags("LDFLAGS", "-ltinfo")
-            env.append_flags("LDFLAGS", "-lz")
```

Comment:
I think we should keep adding `-lz` to `LDFLAGS`. This is unrelelated to the upstream patch, which is concerned only with Curses. Otherwise, I cannot build on macos. I have added one commit bringing back this hunk here: https://github.com/aumuell/spack/tree/ispc_drop_workaround - unfortunately, I don't know how to add this here.

Reply:
The usage of `-lz` is also incorrect, but I haven't had a chance to patch that upstream. I will drop it from this PR.

Reply:
I agree that `-lz` is wrong in general. But it helps. In my opinion, using `llvm-config` to find its dependencies would be the way to go, like this: https://github.com/aumuell/ispc/tree/llvm-config - what do you think?

Reply:
I think that makes sense so long as it is true that ispc only needs those components when they are needed by LLVM. 

In the Spack recipe and in the previous CMake code it read like ncurses is a direct requirements for ISPC.  However, for LLVM, it seems on only depend on ncurses when LLDB is enabled, which is not a requirement by ISPC.